misc(notary): update doc, docker, tee, ci (#874)
tlsnotary/tlsn: crates/notary/server/README.md

      <li><a href="healthcheck">health check</a></li>
      <li><a href="info">info</a></li>
    </ul>
  </body>

concurrency: 32

notarization:
  max_sent_data: 4096
  max_recv_data: 16384
  timeout: 1800
  private_key_path: null
  signature_algorithm: secp256k1

tls:
  enabled: false
  private_key_path: null
  certificate_path: null

log:
  level: DEBUG
  filter: null
  format: COMPACT

auth:
  enabled: false
  whitelist_path: null
```
⚠️ By default, `notarization.private_key_path` is `null`, which means a **random, ephemeral** signing key will be generated at runtime (see [Signing](#signing) for more details).

### Overriding default
The default setting can be overriden with either (1) environment variables, or (2) a configuration file (yaml).

#### Environment Variables
This will override each corresponding default value. It uses the prefix `NS_` followed by the configuration key(s) in uppercase. Double underscores are used for nested configuration keys, e.g. `tls.enabled` will be `NS_TLS__ENABLED`.

Example:
```bash
NS_PORT=8080 NS_NOTARIZATION__MAX_SENT_DATA=2048 cargo run --release
```

#### Configuration File

> This will override all the default values, hence it needs to **contain all compulsory** configuration keys and values (refer to the [default yaml](#default)). This will take precedence over environment variables.

Reviewer: ```suggestion
This will override all the default values, hence it needs to **contain all compulsory** configuration keys and values (refer to the [default yaml](#default)). The config file has precedence over environment variables.
```

String

---
## Features
### Notarization Configuration
To perform a notarization, some parameters need to be configured by the prover and the notary server (more details in the [OpenAPI specification](./openapi.yaml)), i.e.
- maximum data that can be sent and received.
- unique session id.

To streamline this process, a single HTTP endpoint (`/session`) is used by both TCP and WebSocket clients.

### Notarization
After calling the configuration endpoint above, the prover can proceed to start the notarization. For a TCP client, that means calling the `/notarize` endpoint using HTTP, while a WebSocket client should call the same endpoint but using WebSocket. Example implementations of these clients can be found in the [integration test](../tests-integration/tests/notary.rs).

### Signing
To sign the notarized transcript, the notary server requires a signing key. If this signing key (`notarization.private_key_path` in the config) is not provided by the user, then **by default, a random, ephemeral** signing key will be generated at runtime. 

> This ephemeral key, along with its public key, are not saved anywhere by default. They will disappear once the server stops. This makes them only suitable for testing, unless relevant public key infrastructure is used.

Reviewer: ```suggestion
This ephemeral key, along with its public key, is not persisted. The keys disappear once the server stops. This makes the keys only suitable for testing.
```

### TLS
TLS needs to be turned on between the prover and the notary for security purposes. It can be turned off though, if any of the following is true.

1. This server is run locally.
2. TLS is to be handled by an external environment, e.g. reverse proxy, cloud setup.

The toggle to turn on TLS, as well as paths to the TLS private key and certificate can be defined in the config (`tls` field).

### Authorization
An optional authorization module is available to only allow requests with a valid API key attached in the custom HTTP header `X-API-Key`. The API key whitelist path, as well as the flag to enable/disable this module, can be changed in the config (`authorization` field).

> Hot reloading of the whitelist is supported, i.e. modification of the whitelist file will be automatically applied without needing to restart the server. Please take note of the following.
> - Avoid using auto save mode when editing the whitelist to prevent spamming hot reloads.
> - Once the edit is saved, ensure that it has been reloaded successfully by checking the server log.

Reviewer: I'd keep it simpler here
```suggestion
Hot reloading of the whitelist is supported, i.e. changes to the whitelist file are automatically applied without needing to restart the server.
```
